### PR TITLE
Dogfood: Automate dependency checking

### PR DESCRIPTION
**Pull request by dotemacs**
tl;dr I created a workflow that checks the dependencies of the project
using depot and creates a PR to update the dependencies.

I figured that it would be nice for depot to keep an eye on itself and
check if any of its dependencies get stale.

I don't know what your feelings are towards GitHub actions, but seeing
how the project is on GitHub it made sense to me to create an
automated workflow which would check the dependencies of depot, using
depot.

In the PR I've included the steps for creating the badge that can be
displayed in the README showing if the dependencies are up to date or
not. But before I go and do that, do is this something you think that
the project could use?

If you like this change, maybe I can add one that runs tests also,
that way the code can be checked each time somebody makes a PR or you
merge into master.

What do you think?

### CODE REVIEW

**Comment by Olical** [comment]
Ah neat, sorry I missed this, I can't see any harm in checking this, seems pretty useful. Not so sure about the badge part, I don't particularly care about keeping everything up to date unless it's for security or compatibility reasons. So just having it run and check that it exits nicely / doesn't explode is a good canary. Thanks!

**Comment by dotemacs** [comment]
Thanks :)
